### PR TITLE
Launchpad: Enable to use feature flag by setup flow and add flag to the start-writing flow

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to https://github.com/Automattic/wp-calypso/issues/86647

## Proposed Changes
* Enable to use feature flag by setup flow 
* Add feature flag to the `start-writing` flow
* Use a global feature flag to turn off all flows
* The code will check the flags `"launchpad/new-task-definition-parser` and ` "launchpad/new-task-definition-parser/[FLOW_NAME]'

## Testing Instructions
- Load  the setup/free flow  (E.g. /setup/free/launchpad?siteSlug=gabrielcaires-free-flow.wordpress.com)
- Check if there is messages on console log lke the following ones.
``` 
Using new task definition parser {taskId: 'plan_selected', flowId: 'free'}
Using new task definition parser {taskId: 'setup_free', flowId: 'free'}
Using new task definition parser {taskId: 'design_selected', flowId: 'free'}
Using new task definition parser {taskId: 'domain_upsell', flowId: 'free'}
Using new task definition parser {taskId: 'first_post_published', flowId: 'free'}
Using new task definition parser {taskId: 'design_edited', flowId: 'free'}
Using new task definition parser {taskId: 'site_launched', flowId: 'free'}
``` 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
- [ ]

### CODE REVIEW

**Comment by markbiek** [comment]
I'm wondering if it might be better to split up the feature flag changes for the two flows into separate PRs?

Also, there's an `npm` command in Calypso that lets you search for a particular feature flag and it shows you which environments the flag is enabled in.

For example, if I run `npm run feature-search 'launchpad/new-task-definition-parser/free'`, I'll see output like this:

CleanShot 2024-01-18 at 16 39 22@2x

It can be nice to add calling `npm run feature-search <whatever>` to your testing instructions so people can quickly verify the different environments.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/index.ts [hunk 1]
```diff
@@ -16,14 +16,13 @@ const DEFINITIONS = {
 	...planActions,
 } satisfies TaskActionTable;
 
-const MIGRATED_FLOWS = [ 'free' ];
+export const FLAG_NAME = 'launchpad/new-task-definition-parser';
```
Comment (daledupreez) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/index.ts]: Nit: this constant name is not very descriptive. What is the flag used for? What does it do? As an example, calling it `NEW_TASK_DEFINITION_PARSER_FEATURE_FLAG` makes it _much_ clearer what the flag controls, just by reading the name. (We could use something else, too, this is mostly meant to be an illustration/suggestion.) 
```suggestion
export const NEW_TASK_DEFINITION_PARSER_FEATURE_FLAG = 'launchpad/new-task-definition-parser';
```

Comment (gabrielcaires) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/index.ts]: Fixed here https://github.com/Automattic/wp-calypso/pull/86594/commits/04d7c859eb458693c33d8ffb6018594d0750a712

**Comment by gabrielcaires** [comment]
@markbiek  
> I'm wondering if it might be better to split up the feature flag changes for the two flows into separate PRs?

I don't agree much if the idea of add a feature flag without a minimum use case, since this PR is already very small I wondering if the cost benefit of have a PR to only add a line with no logic is good.

**Comment by markbiek** [comment]
> I don't agree much if the idea of add a feature flag without a minimum use case, since this PR is already very small I wondering if the cost benefit of have a PR to only add a line with no logic is good.

@gabrielcaires I understand your point 😁 . If you feel strongly about leaving as-is, that's fine.

My thinking was, what if we had to quickly revert the feature flag _just_ for the `start-writing` flow? If that change was in a separate PR, it'd be very easy to just revert that one PR instead of having to make new code changes.

Worth mentioning is that it's pretty common to have PRs that are just feature flag changes (#82817 and #75574 are a couple of examples. there are [lots of others](https://github.com/Automattic/wp-calypso/pulls?q=is%3Apr%20is%3Aclosed%20feature%20flag%20project%3Aautomattic%2F148).

**Comment by gabrielcaires** [comment]
> @gabrielcaires I understand your point 😁 . If you feel strongly about leaving as-is, that's fine.

@markbiek  I think we are aiming for the same point but with a different approach

 I can see a lot of value in a PR just to ENABLING the feature flag in production, it is the way I am getting used,  as you did on this PR  https://github.com/Automattic/wp-calypso/pull/82817 
 
 I am planning to follow the same approach regardless of when the feature flag was created.

But I do not understand what is the value of a PR (Like this https://github.com/Automattic/wp-calypso/pull/75574/files)  because the default value of `isEnabled` is already false and if we try to revert this PR, it will not cause any effect. 

Could you provide me with more context about it?  It is not only for this PR, it is part of my onboarding :)

**Comment by markbiek** [comment]
> But I do not understand what is the value of a PR (Like this https://github.com/Automattic/wp-calypso/pull/75574/files) because the default value of isEnabled is already false and if we try to revert this PR, it will not cause any effect.

You make a very good point here! I think, in this case, you're correct and it's better to leave this PR the way it is.

**Review by markbiek — APPROVED** [review]
LGTM!

**Pull request merged**